src/routes/safe/store/actions/fetchTransactions.js

```diff
@@ -193,10 +247,43 @@ export const loadSafeTransactions = async (safeAddress: string) => {
   const txsRecord = await Promise.all(
     transactions.map((tx: TxServiceModel) => buildTransactionFrom(safeAddress, tx, safeSubjects)),
   )
+
   return Map().set(safeAddress, List(txsRecord))
 }
 
-export default (safeAddress: string) => async (dispatch: ReduxDispatch<GlobalState>) => {
+export const loadSafeIncomingTransactions = async (safeAddress: string) => {
+  const url = buildIncomingTxServiceUrl(safeAddress)
+  const response = await axios.get(url)
+  const incomingTransactions: IncomingTxServiceModel[] = response.data.results
+  const incomingTxsRecord = await Promise.all(incomingTransactions.map(buildIncomingTransactionFrom))
+
+  return Map().set(safeAddress, List(incomingTxsRecord))
+}
+
+export default (safeAddress: string) => async (dispatch: ReduxDispatch<GlobalState>, getState) => {
   const transactions: Map<string, List<Transaction>> = await loadSafeTransactions(safeAddress)
-  return dispatch(addTransactions(transactions))
+  const incomingTransactions: Map<string, List<IncomingTransaction>> = await loadSafeIncomingTransactions(safeAddress)
+  const state = getState()
+  const storedIncoming = state.incomingTransactions.get(safeAddress) || Map()
+  const mergedIncomingTxs = storedIncoming.merge(incomingTransactions.get(safeAddress))
+
+  if (!mergedIncomingTxs.equals(storedIncoming)) {
```

Comment:
please move this logic to notifications redux middleware

Reply:
actually this isn't merged yet :D but you can see how it's done here https://github.com/gnosis/safe-react/pull/330/files

Reply:
> Also, as far as I can tell the snackbar is only shown when the user is in the transactions tab. Is it also possible to show it on the other tabs?

@lukasschor, done.

Also fixed the outgoing transactions' order.

Thanks for the feedback :smiley: